src/SourceBuild/content/eng/tools/tasks/Microsoft.DotNet.SourceBuild.Tasks.LeakDetection/CustomAttributeTypeProvider.cs

```diff
@@ -0,0 +1,97 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Reflection;
+using System.Reflection.Metadata;
+using System.Reflection.Metadata.Ecma335;
+
+namespace Microsoft.DotNet.SourceBuild.Tasks.LeakDetection
+{
+    internal class CustomAttributeTypeProvider : ICustomAttributeTypeProvider<Type>
+    {
+        private readonly MetadataReader _reader;
+
+        public CustomAttributeTypeProvider(MetadataReader reader)
+        {
+            _reader = reader;
+        }
```

Comment:
This class does not need the `MetadataReader` at construction and can be made a singleton.

```suggestion
        public static readonly CustomAttributeTypeProvider Instance = new();
```

Reply:
Implemented! Thanks for your comments.